extensions/amp-font/0.1/amp-font.js

```diff
@@ -0,0 +1,304 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+/**
+ * @fileoverview Delay loads custom fonts on AMP pages.
+ * The data-shortcode attribute can be easily copied from a normal instagram
+ * URL.
+ * Example:
+ * <code>
+ * <amp-font
+ *   font-family="My Font"
+ *   timeout="3s"
+ *   font-weight="bold"
+ *   on-loading-class="myfont-font-loading"
+ *   on-error-class="myfont-font-missing"
+ *   on-load-class="myfont-font-loaded">
+ * </amp-font>
+ * </code>
+ *
+ * the amp-font element's layout type is nodisplay.
+ */
+
+
+import * as style from '../../../src/style';
+
+/** @private @const {Array.<string>} */
+const DEFAULT_FONTS_ = ['sans-serif', 'serif'];
+
+/** @private @const {string} */
+const TEST_STRING_ = 'MAxmTYklsjo190QW';
+
+/** @private @const {number} */
+const TOLERANCE_ = 2;
+
+/** @private @const {number} */
+const TIMEOUT_ = 5000;
```

Comment:
So, the default timeout in all browsers but Safari is 3s. That is probably a good value here.